src/traces/streamtube/calc.js

```diff
@@ -61,13 +99,18 @@ module.exports = function calc(gd, trace) {
     var filledX;
     var filledY;
     var filledZ;
-    var firstX;
-    var firstY;
-    var firstZ;
+    var firstX, lastX;
+    var firstY, lastY;
+    var firstZ, lastZ;
     if(len) {
-        firstX = x[0];
-        firstY = y[0];
-        firstZ = z[0];
+        firstX = +x[0];
```

Comment:
Again, why not convert all the x, y, z items to numbers first thing in `calc`, so that we don't have to worry about casting strings to numbers in `processGrid`, `filter` and `lessThan`

Reply:
Done in https://github.com/plotly/plotly.js/pull/4431/commits/df9b4c49e62f9fc44624e3fd5f7c37f6bafb1acd.